src/features/tracks/components/TrackUploadModal.tsx

```diff
@@ -1,128 +1,120 @@
+
 
 interface TrackUploadModalProps {
   isOpen: boolean;
   onClose: () => void;
-  onUpload: (file: File) => Promise<void>;
-  trackToUpload: Track | null; 
+  onUpload: (file: File) => void;
+  trackToUpload: Track | null;
   isLoading: boolean;
 }
 
 const TrackUploadModal: React.FC<TrackUploadModalProps> = ({
-    isOpen, onClose, onUpload, trackToUpload, isLoading
+  isOpen,
+  onClose,
+  onUpload,
+  trackToUpload,
+  isLoading,
 }) => {
-    const [selectedFile, setSelectedFile] = useState<File | null>(null);
-    const [previewUrl, setPreviewUrl] = useState<string | null>(null);
-    const [error, setError] = useState<string | null>(null);
-    const audioRef = useRef<HTMLAudioElement>(null);
-
-    useEffect(() => {
-        let objectUrl: string | null = null;
+  const [selectedFile, setSelectedFile] = useState<File | null>(null);
+  const [previewUrl, setPreviewUrl] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
 
-        if (selectedFile) {
-            objectUrl = URL.createObjectURL(selectedFile);
-            setPreviewUrl(objectUrl);
-            console.log("Created preview URL:", objectUrl);
-        } else {
-            setPreviewUrl(null);
-        }
+  useEffect(() => {
+    if (!isOpen) {
+      setSelectedFile(null);
+      setPreviewUrl(null);
+      setError(null);
+    }
+  }, [isOpen]);
 
-        return () => {
-            if (objectUrl) {
-                console.log("Revoking preview URL:", objectUrl);
-                URL.revokeObjectURL(objectUrl);
-            }
-        };
-    }, [selectedFile]); 
+  useEffect(() => {
+    if (selectedFile) {
+      const objectUrl = URL.createObjectURL(selectedFile);
+      setPreviewUrl(objectUrl);
+      return () => {
+        URL.revokeObjectURL(objectUrl);
+      };
+    }
+    setPreviewUrl(null);
+    return () => {}; 
+  }, [selectedFile]);
 
-    const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
-        setError(null);
-        setSelectedFile(null); 
-        if (audioRef.current) {
-            audioRef.current.pause();
-            audioRef.current.currentTime = 0;
-        }
+  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setError(null);
+    const file = e.target.files?.[0];
 
-        if (e.target.files && e.target.files[0]) {
-            const file = e.target.files[0];
-            const allowedTypes = ['audio/mpeg', 'audio/wav', 'audio/ogg'];
-            const maxSizeInBytes = 10 * 1024 * 1024;
-            if (!allowedTypes.includes(file.type)) {
-                setError(`Invalid file type...`);
-                e.target.value = '';
-                return;
-            }
-            if (file.size > maxSizeInBytes) {
-                setError(`File is too large...`);
-                e.target.value = '';
-                return;
-            }
-            setSelectedFile(file);
-        }
-    };
+    if (file) {
```

Comment:
```suggestion
    if (!file) {
      return;
```
Щоби уникнути вложених `if`- ів можна робити early [return](https://medium.com/swlh/return-early-pattern-3d18a41bba8)

Reply:
+